base/windows/FieldPsynergyWindow.js

```diff
@@ -0,0 +1,59 @@
+import { Window } from '../Window.js';
+import { capitalize, get_text_width } from "../../utils.js";
+import * as numbers from '../../magic_numbers.js';
+
+const BASE_WIDTH = numbers.WINDOW_PADDING_H*2 + 8;
```

Comment:
Every time you work with multiplications and divisions with multiples of two with non-floating numbers, use the shift operator: `a * 2` = `(a << 1)`, `a * 4` = `(a << 1)`, `a / 2` = `(a >> 1)`...